CONTRIBUTING.md

```diff
@@ -75,6 +75,24 @@ Configure your editor to run `cargo fmt` on save. Also configure it to strip tra
 
 To check everything in one go, run `./scripts/check.sh`. `check.sh` should ideally check approximately the same things as our CI.
 
+### Linting
+Prior to pushing changes to a PR, at a minimum, you should always run `just fast-lint`. This is designed to run
+in a few seconds and should catch the more trivial issues to avoid wasting CI time.
+
+### Hooks
+We recommend adding the rerun pre-push hook to your local checkout, which among other-things will run
+`just fast-lint` for you.
+
+To install the hooks, simply copy them into the `.git/hooks` directory of your local checkout.
+```
+cp hooks/pre-push .git/hooks/pre-push
+chmod +x .git/hooks/pre-push
```

Comment:
why not `chmod +x hooks/pre-push` already?

Reply:
It generally should be, though I've definitely seen edge-cases across OS's and git configurations where that isn't guaranteed.